### PR TITLE
SAM.gov integration ADR

### PR DESCRIPTION
**Pull request by jadudm**
Acknowledges the current state of systems, our approach to integration today, and our paths forward in the future.

### CODE REVIEW

File: docs/architecture/decisions/0017-samgov-integration.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+User enters UEI       │               │
+                      └──────┬────────┘
+                             │
+                             │           Do not pre-populate fields,
+                             └────────►  Mark UEI as not validated.
+                             Not found
+
+UEIs will always be validated. This means we will determine that they have the correct shape.
+
+When a submitter enters their UEI, we will attempt to validate it with the SAM.gov API. If we can, we will pre-populate the entity name with the information returned. We will also record that the UEI was validated at time of submission.
```
[the remark this answers is not in the document]
Comment (jadudm) [reply docs/architecture/decisions/0017-samgov-integration.md:30]: I think it's a correct use of "verify" and "validate," but we could back those words out, and just use the plain language descriptions. 

I'll update and remove the language that requires definition. I think that simplifies the document.

File: docs/architecture/decisions/0017-samgov-integration.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+If we cannot validate the UEI, we will record the UEI and the fact that we could not validate it at the time of submission.
+
+In the submission process, there is a later point where the UEI might be edited. If it is, we will (at that point) allow the edit to take place, but we will mark the UEI as non-validated.
```
[the remark this answers is not in the document]
Comment (jadudm) [reply docs/architecture/decisions/0017-samgov-integration.md:34]: ... yes. Thank you. Updating.

File: docs/architecture/decisions/0017-samgov-integration.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jadudm) [reply docs/architecture/decisions/0017-samgov-integration.md]: Added note about correct shape or empty string.

**Comment by jadudm** [comment]
Title updated.

**Review by tadhg-ohiggins — APPROVED** [review]

**Review by mogul — APPROVED** [review]

**Pull request merged**